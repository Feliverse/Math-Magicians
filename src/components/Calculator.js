import React from 'react';
import Display from './Display';
import Buttons from './Button';

const Calculator = () => (
  <section className="calculator-container">
    <Display />
    <Buttons />
  </section>
);

export default Calculator;
