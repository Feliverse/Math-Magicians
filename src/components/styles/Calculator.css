.calculator {
  font-family: 'Times New Roman', Times, serif;
  padding: 2rem 1rem;
  height: 80vh;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  background-color: #f0f8fff5;
}

@media only screen and (min-width: 600px) {
  .calculator {
    flex-direction: row;
    justify-content: space-around;
    align-items: flex-start;
  }
}
