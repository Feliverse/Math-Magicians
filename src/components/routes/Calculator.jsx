import '../styles/Calculator.css';
import Calculator from '../Calculator';

const CalculatorPage = () => (
  <section className="calculator">
    <h2>Lets do some math!</h2>
    <Calculator />
  </section>
);

export default CalculatorPage;
