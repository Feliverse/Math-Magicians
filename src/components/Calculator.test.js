import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Calculator from './Calculator';

test('renders Calculator component', () => {
  const { asFragment } = render(<Calculator />);
  expect(asFragment()).toMatchSnapshot();
});

test('User click on buttons', () => {
  const handleClick = jest.fn();
  render(<button type="button" onClick={handleClick} className="btn">AC</button>);
  const button = screen.getByText('AC');
  userEvent.click(button);
  expect(handleClick).toHaveBeenCalled();
});