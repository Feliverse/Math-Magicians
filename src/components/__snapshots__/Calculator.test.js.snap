// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator component 1`] = `
<DocumentFragment>
  <section
    class="calculator-container"
  >
    <div
      class="calculator-screen"
    >
      <span
        class="calculator-display"
      >
        0
      </span>
    </div>
    <div
      class="calculator-keypad"
    >
      <button
        class="btn"
        type="button"
      >
        AC
      </button>
      <button
        class="btn"
        type="button"
      >
        +/-
      </button>
      <button
        class="btn"
        type="button"
      >
        %
      </button>
      <button
        class="btn orange"
        type="button"
      >
        รท
      </button>
      <button
        class="btn"
        type="button"
      >
        7
      </button>
      <button
        class="btn"
        type="button"
      >
        8
      </button>
      <button
        class="btn"
        type="button"
      >
        9
      </button>
      <button
        class="btn orange"
        type="button"
      >
        x
      </button>
      <button
        class="btn"
        type="button"
      >
        4
      </button>
      <button
        class="btn"
        type="button"
      >
        5
      </button>
      <button
        class="btn"
        type="button"
      >
        6
      </button>
      <button
        class="btn orange"
        type="button"
      >
        -
      </button>
      <button
        class="btn"
        type="button"
      >
        1
      </button>
      <button
        class="btn"
        type="button"
      >
        2
      </button>
      <button
        class="btn"
        type="button"
      >
        3
      </button>
      <button
        class="btn orange"
        type="button"
      >
        +
      </button>
      <button
        class="btn double"
        type="button"
      >
        0
      </button>
      <button
        class="btn"
        type="button"
      >
        .
      </button>
      <button
        class="btn orange"
        type="button"
      >
        =
      </button>
    </div>
  </section>
</DocumentFragment>
`;
